### DISCUSSION TITLE
Making a compact/dense Table

### DISCUSSION THREAD

**Discussion by troypoulter** [zeit-ui/react#631] [Q&A]
I'm using the [Table](https://react.geist-ui.dev/en-us/components/table) component and I'm trying to make it more compact/dense by ideally:

- Reducing the height to only cover the text and a smaller margin.
- Potentially make the font smaller.

I _think_ using [TableRowClassNameHandler](https://react.geist-ui.dev/en-us/components/table#tablerowclassnamehandler) is the go but am unsure of how best to implement it and I didn't see it used in one of the examples for reference.

**Comment by harshcut** [comment]
If you want to make everything smaller, try `scale`.

```jsx
<Table data={data} scale={0.5}>
  <Table.Column prop="property" label="property" />
  <Table.Column prop="description" label="description" />
  <Table.Column prop="type" label="type" />
  <Table.Column prop="default" label="default" />
</Table>
```

**Reply by troypoulter** [reply]
Thanks for the suggestion! I can see when I updated the sample code in https://react.geist-ui.dev/en-us/components/table with the `scale` it worked as expected.

I then tried applying it to my React app which uses Typescript I get the below error:

```
Type '{ children: Element[]; data: Result[] | undefined; scale: number; }' is not assignable to type 'IntrinsicAttributes &
Pick<PropsWithChildren<TableProps<Result>>, "data" | "color" | "defaultChecked" | "defaultValue"
| ... 255 more ... | "cellSpacing"> & Partial<...> & Partial<...>'.
  Property 'scale' does not exist on type 'IntrinsicAttributes &
Pick<PropsWithChildren<TableProps<Result>>, "data" | "color" | "defaultChecked"
| "defaultValue" | ... 255 more ... | "cellSpacing"> & Partial<...> & Partial<...>'.  TS2322

    39 |                     </Select>
    40 |                     <Spacer h={2} />
  > 41 |                     <Table data={systems?.results} scale={0.5}>
       |                                                    ^
    42 |                         <Table.Column prop='name' label='Name' />
```

I'm using `"@geist-ui/react": "^2.2.0"` which I believe is the latest version atm.

I'm able to use other components all good with `scale` like `Text` & `Select` but Typescript is throwing issues when I try to use it on the `Table`.

**Reply by harshcut** [reply]
Maybe @unix can help you here. For now `@ts-ignore` should remove the error.

```jsx
return (
  <div className="App">
    {/* @ts-ignore */}
    <Table data={data} scale={0.5}>
      <Table.Column prop="property" label="property" />
      <Table.Column prop="description" label="description" />
      <Table.Column prop="type" label="type" />
      <Table.Column prop="default" label="default" />
    </Table>
  </div>
);
```

**Reply by troypoulter** [reply]
Thank you for the workaround!

**Reply by troypoulter** [reply]
On a similar topic with using `scale` to reduce the whole table which works well, what if I wanted to make the rows narrower so they don't have as much white space above and below (red arrows).

![image](https://user-images.githubusercontent.com/19419349/133726622-3bc6caae-9585-469b-b752-7104a64f35bf.png)

I see in the docs there is props to change the width of the column and thinking I could use `rowClassName` prop but is there a simple way to reduce that white space so it can be more dense, without needing to reduce the text size.

**Reply by harshcut** [reply]
Apply `min-height` property for `.thread-box` and `.cell`. This worked for me in Next.js.

```jsx
:global(.thead-box) {
  min-height: 1rem !important;
}

:global(.cell) {
  min-height: 1rem !important;
}
```

**Reply by troypoulter** [reply]
Is there a way to only apply that to tables I specify? I'm still new to this type of styling so unsure how best to set it for some tables and not all. Thank you for the continued support!! 🎉

**Reply by harshcut** [reply]
That should be easy. Try separating them into different components. Create a wrapper component for each `Table` for which you want a different style.
